pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -61,14 +61,15 @@ func (s *preFilterState) Clone() framework.StateData {
 		// Constraints are shared because they don't change.
 		Constraints:          s.Constraints,
 		TpKeyToCriticalPaths: make(map[string]*criticalPaths, len(s.TpKeyToCriticalPaths)),
-		TpPairToMatchNum:     make(map[topologyPair]int32, len(s.TpPairToMatchNum)),
+		TpPairToMatchNum:     make(map[topologyPair]*int32, len(s.TpPairToMatchNum)),
 	}
 	for tpKey, paths := range s.TpKeyToCriticalPaths {
 		copy.TpKeyToCriticalPaths[tpKey] = &criticalPaths{paths[0], paths[1]}
 	}
 	for tpPair, matchNum := range s.TpPairToMatchNum {
 		copyPair := topologyPair{key: tpPair.key, value: tpPair.value}
-		copy.TpPairToMatchNum[copyPair] = matchNum
+		copyCount := *matchNum
+		copy.TpPairToMatchNum[copyPair] = &copyCount
```

Comment:
I believe we should also consider the performance of Clone() as Clone() is used a lot during preemption. Could you benchmark it? (we can tolerate slight perf downgrade, but not much)

I asked this as instantiating a pointer is expensive than we think - we used to hit some regression when submitting performance PRs.

Reply:
I got this:

```
master

BenchmarkFilter/1000nodes/single-constraint-zone/Clone-56   	  269688	      4880 ns/op
BenchmarkFilter/1000nodes/single-constraint-node/Clone-56   	    4214	    295056 ns/op
BenchmarkFilter/1000nodes/two-Constraints-zone-node/Clone-56         	    3637	    305096 ns/op

PR

BenchmarkFilter/1000nodes/single-constraint-zone/Clone-56   	  254336	      5195 ns/op
BenchmarkFilter/1000nodes/single-constraint-node/Clone-56   	    3175	    380806 ns/op
BenchmarkFilter/1000nodes/two-Constraints-zone-node/Clone-56         	    2961	    369687 ns/op
```

Which translates to 23% slower for 1000 nodes. Given that we are still talking about less than 1ms, I'm ok with this. If you agree, I'll include this numbers in the PR description.

Reply:
I'm fine with it - leave a comment in the code is also fine.